drivers/espi/espi_mchp_xec.c

```diff
@@ -0,0 +1,822 @@
+				struct espi_oob_packet pckt)
+{
+	return -ENOTSUP;
+}
+
+static int espi_xec_receive_oob(struct device *dev,
+				struct espi_oob_packet pckt)
+{
+	return -ENOTSUP;
+}
+
+static int espi_xec_flash_read(struct device *dev,
+				struct espi_flash_packet pckt)
+{
+	return -ENOTSUP;
+}
+
+static int espi_xec_flash_write(struct device *dev,
+				struct espi_flash_packet pckt)
+{
+	return -ENOTSUP;
+}
+
+static int espi_xec_flash_erase(struct device *dev,
+				struct espi_flash_packet pckt)
+{
+	return -ENOTSUP;
+}
+
+static int espi_xec_manage_callback(struct device *dev,
+					struct espi_callback *callback,
+					bool set)
+{
+	struct espi_xec_data *data =
+		(struct espi_xec_data *) (dev->driver_data);
+
+	return espi_manage_callback(&data->callbacks, callback, set);
+}
+
+static void send_slave_bootdone(struct device *dev)
+{
+	int ret;
+	u8_t boot_done;
+
+	ret = espi_xec_receive_vwire(dev, ESPI_VWIRE_SIGNAL_SLV_BOOT_DONE,
+				&boot_done);
+	if (!ret && !boot_done) {
+		/* SLAVE_BOOT_DONE & SLAVE_LOAD_STS have to be sent together */
+		ESPI_S2M_VW_REGS->SMVW01.SRC = 0x01000001;
+	}
+}
+
+#ifdef CONFIG_ESPI_CHANNEL_OOB
+static void espi_init_oob(struct device *dev)
+{
+	struct espi_xec_config *config =
+		(struct espi_xec_config *) (dev->config->config_info);
+
+	MCHP_GIRQ_ENSET(config->bus_girq_id) =
+		BIT(MCHP_ESPI_OOB_UP_GIRQ_POS) |
+		BIT(MCHP_ESPI_OOB_DN_GIRQ_POS);
+
+}
+#endif
+#ifdef CONFIG_ESPI_CHANNEL_FLASH
+static void espi_init_flash(struct device *dev)
+{
+	struct espi_xec_config *config =
+		(struct espi_xec_config *) (dev->config->config_info);
+
+	MCHP_GIRQ_ENSET(config->bus_girq_id) = BIT(MCHP_ESPI_FC_GIRQ_POS);
+}
+#endif
+
+static void espi_rst_isr(struct device *dev)
+{
+	struct espi_xec_data *data =
+		(struct espi_xec_data *) (dev->driver_data);
+	struct espi_event evt = { ESPI_BUS_RESET, 0, 0};
+
+	if (ESPI_CAP_REGS->ERST_STS & MCHP_ESPI_RST_ISTS) {
+		ESPI_CAP_REGS->ERST_STS |= MCHP_ESPI_RST_ISTS;
+
+		/* TODO: Check if we need to detect eSPI reset after boot */
+		ESPI_CAP_REGS->ERST_IEN = 0;
+		if (ESPI_CAP_REGS->ERST_STS & ~MCHP_ESPI_RST_ISTS_PIN_RO_HI) {
+			data->espi_rst_asserted = 1;
+		} else {
+			data->espi_rst_asserted = 0;
+		}
+	evt.evt_data = data->espi_rst_asserted;
+	espi_send_callbacks(&data->callbacks, dev, evt);
+
+#ifdef CONFIG_ESPI_CHANNEL_OOB
+	espi_init_oob(dev);
+#endif
+#ifdef CONFIG_ESPI_CHANNEL_FLASH
+	espi_init_flash(dev);
+#endif
+	}
+}
+
+static void config_sub_devices(void)
+{
+	KBC_REGS->KBC_CTRL |= MCHP_KBC_CTRL_AUXH;
+	KBC_REGS->KBC_PORT92_EN |= MCHP_KBC_PORT92_EN;
+	ESPI_EIO_BAR_REGS->EC_BAR_KBC |= MCHP_ESPI_IO_BAR_HOST_VALID;
+	ESPI_EIO_BAR_REGS->EC_BAR_ACPI_EC_0 |= MCHP_ESPI_IO_BAR_HOST_VALID;
+	ESPI_EIO_BAR_REGS->EC_BAR_ACPI_EC_1 = 0x06A00001;
+	ESPI_EIO_BAR_REGS->EC_BAR_MBOX = 0x03600001;
+	ESPI_EIO_BAR_REGS->EC_BAR_PORT92 = 0x00920001;
+	ESPI_EIO_BAR_REGS->EC_BAR_UART_0 = 0x03F80001;
+	ESPI_EIO_BAR_REGS->EC_BAR_P80CAP_0 = 0x00810001;
```

Comment:
All there BARs should come from DTS only.

Btw, you have ESPI controller device driver (and dts bindings), do you need any espi device then? (in dts you would have a espi device with an espi controller parent then).